### ISSUE TITLE
Precompiling failure for Singular

### ISSUE THREAD

**Issue by wagh** [oscar-system/Singular.jl#618]
I have Ubuntu 21.10. I have installed julia from the [official download page](https://julialang.org/downloads/) -- Generic Linux x86.

When I try 
      julia> using Pkg; Pkg.add("Singular")
      julia> using Singular
I am getting an error, probably with libsingular (error log attached).

Am I missing something?

-- VInay

[julia-install-error.txt](https://github.com/oscar-system/Singular.jl/files/10688339/julia-install-error.txt)

**Comment by thofma** [comment]
Could you post the output of
```
julia> import Pkg; Pkg.status(mode=Pkg.PKGMODE_MANIFEST)
```
here? Thanks

**Comment by wagh** [comment]
Here is the required o/p:

            julia> import Pkg; Pkg.status(mode=Pkg.PKGMODE_MANIFEST)
            Status `~/.julia/environments/v1.8/Manifest.toml`
            ⌅ [c3fe647b] AbstractAlgebra v0.16.0
              [b99e7846] BinaryProvider v0.5.10
            ⌅ [1f15a43c] CxxWrap v0.11.2
            ⌅ [ffbed154] DocStringExtensions v0.8.6
            ⌅ [c863536a] GAP v0.5.2
              [e39c9192] GroebnerBasis v0.3.3
            ⌅ [3e1990a7] Hecke v0.10.5
              [692b3bcd] JLLWrappers v1.4.1
              [682c06a0] JSON v0.21.3
            ⌅ [472f376f] LoadFlint v0.5.1
              [1914dd2f] MacroTools v0.5.10
            ⌅ [4fe8b98c] Mongoc v0.6.2
            ⌅ [2edaba10] Nemo v0.22.1
            ⌃ [f1435218] Oscar v0.5.2
              [69de0a69] Parsers v2.5.5
            ⌅ [d720cf60] Polymake v0.5.6
              [21216c6a] Preferences v1.3.0
              [fb686558] RandomExtensions v0.4.3
              [ae029012] Requires v1.3.0
            ⌅ [bcd08a7b] Singular v0.5.2
              [66db9d55] SnoopPrecompile v1.0.3
              [90137ffa] StaticArrays v1.5.12
              [1e83bf80] StaticArraysCore v1.4.0
            ⌅ [e21ec000] Antic_jll v0.200.400+0
            ⌅ [d9960996] Arb_jll v200.1900.1+0
            ⌅ [e134572f] FLINT_jll v200.700.100+0
            ⌅ [5cd7a574] GAP_jll v400.1100.100+0
            ⌅ [de1ad85e] GAP_lib_jll v400.1100.100+0
              [e8aa6df9] GLPK_jll v5.0.1+0
              [dd4b983a] LZO_jll v2.10.1+0
              [90100e71] MongoC_jll v1.19.1+0
              [68e3532b] Ncurses_jll v6.2.0+0
              [76642167] Ninja_jll v1.11.1+0
              [458c3c95] OpenSSL_jll v1.1.19+0
              [80dd9cbb] PPL_jll v1.2.1+0
            ⌅ [83958c19] Perl_jll v5.30.3+2
              [05236dd9] Readline_jll v8.1.1+1
            ⌅ [43d676ae] Singular_jll v402.0.104+0
              [3161d3a3] Zstd_jll v1.5.2+0
            ⌅ [508c9074] bliss_jll v0.73.1+0
            ⌅ [28df3c45] boost_jll v1.71.0+3
              [f07e07eb] cddlib_jll v0.94.13+0
              [006bdf2e] gb_jll v0.17.0+0
              [1493ae25] lib4ti2_jll v1.6.10+0
            ⌅ [3eaa8342] libcxxwrap_julia_jll v0.8.8+1
            ⌅ [4d8266f6] libpolymake_julia_jll v0.4.117+0
            ⌅ [ae4fbd8f] libsingular_julia_jll v0.9.205+0
            ⌅ [3873f7d0] lrslib_jll v0.2.0+0
              [55c6dc9b] nauty_jll v2.6.13+0
            ⌅ [6690c6e9] normaliz_jll v300.800.901+0
            ⌅ [7c209550] polymake_jll v400.400.0+1
              [fe1e1685] snappy_jll v1.1.9+1
              [0dad84c5] ArgTools v1.1.1
              [56f22d72] Artifacts
              [2a0f44e3] Base64
              [ade2ca70] Dates
              [8ba89e20] Distributed
              [f43a241f] Downloads v1.6.0
              [7b1f6079] FileWatching
              [b77e0a4c] InteractiveUtils
              [b27032c2] LibCURL v0.6.3
              [76f85450] LibGit2
              [8f399da3] Libdl
              [37e2e46d] LinearAlgebra
              [56ddb016] Logging
              [d6f4376e] Markdown
              [a63ad114] Mmap
              [ca575930] NetworkOptions v1.2.0
              [44cfe95a] Pkg v1.8.0
              [de0858da] Printf
              [9abbd945] Profile
              [3fa0cd96] REPL
              [9a3f8284] Random
              [ea8e919c] SHA v0.7.0
              [9e88b42a] Serialization
              [6462fe0b] Sockets
              [2f01184e] SparseArrays
              [10745b16] Statistics
              [fa267f1f] TOML v1.0.0
              [a4e569a6] Tar v1.10.1
              [8dfed614] Test
              [cf7118a7] UUIDs
              [4ec0a83e] Unicode
              [e66e0078] CompilerSupportLibraries_jll v1.0.1+0
              [781609d7] GMP_jll v6.2.1+2
              [deac9b47] LibCURL_jll v7.84.0+0
              [29816b5a] LibSSH2_jll v1.10.2+0
              [3a97d323] MPFR_jll v4.1.1+1
              [c8ffd9c3] MbedTLS_jll v2.28.0+0
              [14a3606d] MozillaCACerts_jll v2022.2.1
              [4536629a] OpenBLAS_jll v0.3.20+0
              [83775a58] Zlib_jll v1.2.12+3
              [8e850b90] libblastrampoline_jll v5.1.1+0
              [8e850ede] nghttp2_jll v1.48.0+0
              [3f19e933] p7zip_jll v17.4.0+0
            Info Packages marked with ⌃ and ⌅ have new versions available, but those with ⌅ are restricted by compatibility constraints from upgrading. To see why use `status --outdated -m`

**Comment by fingolfin** [comment]
You have heavily outdated versions of all packages installed.

Try doing `Pkg.update()`.

It is possible this won't do anything due to conflicting version requirements. In that case I recommend that you remove all packages from tye global Julia environment and then reinstall only those you really use

**Comment by wagh** [comment]
@fingolfin as you expected, `Pkg.update()` didnt help! 

I wonder how these "outdated" packages were installed, as I had a fresh installation of julia and then packages were installed using `Pkg.add(...)`. If this standard process installs outdated and also a few extra packages, then I guess this is also a bug(?) at some level.

**Comment by fingolfin** [comment]
You probably did not delete your `~/.julia` directory so it wasn't a fully "fresh" installation

**Comment by thofma** [comment]
Before deleting `~/.julia`, what does
```
julia> Pkg.pkg"add Oscar@0.11.2"
```
say? Only the error is interesting.

**Comment by wagh** [comment]
Sorry, I have already deleted... :-(. Now the "fresh" installation seems to have stuck at precompiling Singular.

Another experiment I did: On the fresh installation of Julia, I did `Pkg.add( "Sigular" ). ` It works perfectly. 

Then by removing the ~/.julia, I had another "fresh" installation. ON this I did `Pkg.add( "GroebnerBasis" )`. -- at this, the system is taking almost forever to precompile Singular. 

The first time when I reported this, I had added GroebnerBasis before SIngular. 

So now I wonder if the problem is at Singular OR GroebnerBasis!
